templates/server/src/main/resources/archetype-resources/api/src/test/java/__packageInPathFormat__/ToTest.java

```diff
@@ -58,32 +60,30 @@ public void testTos() {
   /**
    * Checks for an overridden equals() method of a given class object
    */
-  private boolean checkForEquals(Class<?> clazz) {
+  private boolean checkForCustomEquals(Class<?> clazz) {
 
-	try {
-	  clazz.getMethod("java.lang.Object.equals", Object.class);
-	  return false;													// No custom implementation
-	} catch(NoSuchMethodException noEquals) {}
+	for(Method method : clazz.getMethods()) {
+	  if(method.toString().equals("public boolean java.lang.Object.equals(java.lang.Object)")) return false;
+	}
 
 	return true;
   }
 
   /**
    * Checks for an overridden hashCode() method of a given class object
    */
-  private boolean checkForHashCode(Class<?> clazz) {
+  private boolean checkForCustomHashCode(Class<?> clazz) {
 
-	try {
-	  clazz.getMethod("java.lang.Object.hashCode");
-	  return false;											// No custom implementation
-	} catch(NoSuchMethodException noHashCode) {}
+	for(Method method : clazz.getMethods()) {
+	  if(method.toString().equals("public native int java.lang.Object.hashCode()")) return false;
+	}
 
 	return true;
   }
 
   private void testEqualsAndHashcode(Class<?> clazz, SoftAssertions assertion, ToExclusion exclusion) {
 
-	if(checkForEquals(clazz) || checkForHashCode(clazz)) {
+	if((checkForCustomEquals(clazz) == false) || (checkForCustomHashCode(clazz) == false)) {
```

Comment:
in general you would do it as !checkForCustomEquals(clazz) rather than comparing the equality.
Also it would be easier to read if you provide a more fluent readable name for the methods. Thus, you would not even had made the error initially with the && ;)
For example call the method `hasCustomEquals`, then the return value is clearly understood.